### ISSUE TITLE
Unable to reproduce results

### ISSUE THREAD

**Issue by yzfly** [arunos728/MotionSqueeze#7]
Can you share your training settings for ResNet50-TSM?  I use following training settings and only get testing Prec@1 47.856.

--arch "MS" --num_segments 8 --mode 1 --gd 200 --lr 0.02 --lr_steps 20 30 --epochs 40 -b 32 -i 1 -j 4 --dropout 0.5

Testing Results: Prec@1 47.856 Prec@5 78.021 Loss 2.20553 Time 0.1840

**Comment by mitming** [comment]
> Can you share your training settings for ResNet50-TSM? I use following training settings and only get testing Prec@1 47.856.
> 
> --arch "MS" --num_segments 8 --mode 1 --gd 200 --lr 0.02 --lr_steps 20 30 --epochs 40 -b 32 -i 1 -j 4 --dropout 0.5
> 
> Testing Results: Prec@1 47.856 Prec@5 78.021 Loss 2.20553 Time 0.1840

hi, I have the same question, have you resolve this? Could you please tell me how to fix it?

**Comment by yzfly** [comment]
> > Can you share your training settings for ResNet50-TSM? I use following training settings and only get testing Prec@1 47.856.
> > --arch "MS" --num_segments 8 --mode 1 --gd 200 --lr 0.02 --lr_steps 20 30 --epochs 40 -b 32 -i 1 -j 4 --dropout 0.5
> > Testing Results: Prec@1 47.856 Prec@5 78.021 Loss 2.20553 Time 0.1840
> 
> hi, I have the same question, have you resolve this? Could you please tell me how to fix it?

No, I also wander how to reproduce the results.

**Comment by arunos728** [comment]
According to the results, it seems that 'Spatial Correlation Sampler' does not work.
We just have provided another matching layer (Matching_layer_mm in resnet_TSM.py) which consists of matrix multiplication & index rearrangement.
Originally, the output of (Matching_layer_scs) and (Matching_layer_mm) should be the same.
You can check out the result of [SpatialCorrelationSampler_and_MatMul.ipynb](https://github.com/arunos728/MotionSqueeze/blob/master/SpatialCorrelationSampler_and_MatMul.ipynb).
If the tensor outputs of two matching layers are not the same, you should re-install Spatial Correlation Sampler based on [environment](https://github.com/arunos728/MotionSqueeze/blob/master/environment.yml) or just use Matching_layer_mm (line 310 instead of line 309 in resnet_TSM.py).
We also provide the instructions for installing Spatial Correlation Sampler'.
